packages/app/cypress/e2e/debug.cy.ts

```diff
@@ -161,12 +136,95 @@ describe('App - Debug Page', () => {
+    cy.findByTestId('run-failures').should('exist').should('have.attr', 'href', '#/specs/runner?file=src/NewComponent.spec.jsx&mode=debug')
 
     cy.findByLabelText('Open in IDE').click()
     cy.wait('@openFileInIDE')
     cy.withCtx((ctx) => {
-      expect(ctx.actions.file.openFile).to.have.been.calledWith('src/components/InfoPanel/InfoPanel.cy.ts', 1, 1)
+      expect(ctx.actions.file.openFile).to.have.been.calledWith('src/NewComponent.spec.jsx', 1, 1)
+    })
+  })
+
+  it('shows running and updating build', () => {
+    cy.remoteGraphQLIntercept((obj, _testState, options) => {
+      if (obj.operationName === 'RelevantRunsDataSource_RunsByCommitShas') {
+        obj.result.data = options.RelevantRunsDataSource_RunsByCommitShas.data
+      }
+
+      const originalRun = options.DebugDataFailing.data.currentProject.cloudProject.runByNumber
+
+      if (options.testRun === undefined) {
+        options.testRun = JSON.parse(JSON.stringify(originalRun))
+      }
+
+      const run = options.testRun
+
+      run.totalInstanceCount = 5
+      if (run.completedInstanceCount === undefined) {
+        run.completedInstanceCount = 0
+        run.createdAt = (new Date()).toISOString()
+      }
+
+      if (run.totalInstanceCount === run.completedInstanceCount) {
+        run.status = 'FAILED'
+      } else {
+        run.status = 'RUNNING'
+      }
+
+      if (run.completedInstanceCount < 3) {
+        run.testsForReview = []
+      } else {
+        run.testsForReview = originalRun.testsForReview
+      }
+
+      run.totalFailed = run.testsForReview.length
+      run.totalPassed = run.completedInstanceCount - run.totalFailed
+
+      if (obj.operationName === 'Debug_currentProject_cloudProject_cloudProjectBySlug') {
+        if (obj.result.data) {
+          obj.result.data.cloudProjectBySlug.runByNumber = run
+        }
+      }
+
+      if (obj.operationName === 'RelevantRunSpecsDataSource_Specs' && obj.result.data) {
+        //NOTE Figure out how to manually trigger polling instead of adjusting polling intervals
+        obj.result.data.pollingIntervals = {
+          __typename: 'CloudPollingIntervals',
+          runByNumber: 1, //Increase polling interval for debugging test
+        }
+
+        if (run.totalInstanceCount === run.completedInstanceCount) {
+          obj.result.data.pollingIntervals.runByNumber = 100
+        } else {
+          run.completedInstanceCount = run.completedInstanceCount !== undefined ? ++run.completedInstanceCount : 0
+        }
+
+        obj.result.data.cloudNodesByIds = [
+          run,
+        ]
+      }
+
+      return obj.result
+    }, { RelevantRunsDataSource_RunsByCommitShas, DebugDataFailing })
+
+    cy.visitApp()
+
+    cy.findByTestId('sidebar-link-debug-page').click()
+    cy.findByTestId('debug-container').should('be.visible')
+
+    cy.findByTestId('header-top').contains('chore: testing cypress')
```

Comment:
In general I would like to add some more visibility assertions to this spec file where we use `contains`, particularly because we do not have any percy snapshots in app e2e tests. 

Without visibility assertions, we are confirming what is in the DOM, but we won't catch problems caused by CSS that might still hide the elements in various ways. 

For example, for this assertion, we are passing at this point, after DOM is updated but before CSS rendering has caught up:

![Screenshot 2023-06-15 at 2 19 04 PM](https://github.com/cypress-io/cypress/assets/8340719/4c1d1813-2bbf-4b03-8e7e-f1d2f691923d)

Reply:
makes me really want to make a `containsVisible` custom command and just never think about this again.